### ISSUE TITLE
ofBuffer clear() removes data but retains RAM

### ISSUE THREAD

**Issue by armadillu** [openframeworks/openFrameworks#6271]
ofBuffer may lead to memory hogging in non obvious ways. I've just spent two hours looking for what I thought to be a memory leak that wasn't. 

It turns out ofBuffer internally uses a `std::vector<T>` to fit data. `std::vector<T>` defaults to a greedy behavior in repsect of memory use; once a vector has been "grown" to a certain size, even if you clear its data back to zero, its internal data structure keeps the memory it took to fill it in the past in reserve, despite reporting a size() of zero. They call this the vector [capacity()](http://www.cplusplus.com/reference/vector/vector/capacity/). 

I have a project with lots of ofBuffers that continiously load and release large amounts of data, and even when they were all empty (with clear()), the RAM usage of the app was still massive. I guess this behavior is ok for most cases, but it did bite me. I see no way to "shrink" the ofBuffer back to a capacity size relative to its current contents. This means that if, let's say, I load 1Gb of data into that ofBufffer, even if I `clear()` that ofBuffer, that 1GB of RAM is still claimed by the ofBuffer and there's no way to claim it back (unless I destroy the ofBuffer object alltogehter).

C++'s std::vector<T> has a method named [shrink_to_fit()](http://www.cplusplus.com/reference/vector/vector/shrink_to_fit/) that would be useful here; I would propose to either expose the internal `vector<T>` member ( [buffer](https://github.com/openframeworks/openFrameworks/blob/master/libs/openFrameworks/utils/ofFileUtils.h#L267) ) in the API, or wrap the `shrink_to_fit()`method and expose that?

**Comment by arturoc** [comment]
Yes that's the way it works for performance reasons. If you are loading a lot of stuff probably reusing ofBuffers from a pool will give you both performance and reasonable memory usage but in any case if you just want to completely reset the buffers to zero you can just reset the whole object doing:

```
buffer = ofBuffer();
```

Another way to ensure that it won't grow more than you need if you know the amount of memory in advance is to `reserve` the memory before loading data so it doesn't grow exponentially and only has the capacity you need

but yes we can definitely add something equivalent to shring_to_fit if you want to send a PR

**Comment by armadillu** [comment]
ok will do soon!
